mpp/shmem.h4.in

```diff
@@ -64,6 +64,7 @@ extern "C" {
 #define  SHMEM_REDUCE_MIN_WRKDATA_SIZE @C_REDUCE_MIN_WRKDATA_SIZE@
 
 #define SHMEM_HAVE_VISIBILITY @SHMEM_C_HAVE_VISIBILITY@
+#define SHMEM_HAVE_ATTRIBUTE_DEPRECATED @SHMEM_HAVE_ATTRIBUTE_DEPRECATED@
```

Comment:
I think these are both attributes?  Should we have a consistent naming scheme for them?

Reply:
I think I slightly prefer SHMEM_HAVE_ATTRIBUTE_...  But, your choice.  🎲 😁